conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,208 @@
+            <Conan{name}BinaryDirectories>{bin_dirs}</Conan{name}BinaryDirectories>
+            <Conan{name}Libraries>{libs}</Conan{name}Libraries>
+            <Conan{name}SystemDeps>{system_libs}</Conan{name}SystemDeps>
+          </PropertyGroup>
+          <PropertyGroup>
+            <LocalDebuggerEnvironment>PATH=%PATH%;$(Conan{name}BinaryDirectories)
+            </LocalDebuggerEnvironment>
+            <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
+          </PropertyGroup>
+          <ItemDefinitionGroup>
+            <ClCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)
+              %(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ClCompile>
+            <Link>
+              <AdditionalLibraryDirectories>$(Conan{name}LibraryDirectories)
+              %(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+              <AdditionalDependencies>$(Conan{name}Libraries)%(AdditionalDependencies)
+              </AdditionalDependencies>
+              <AdditionalDependencies>$(Conan{name}SystemDeps)%(AdditionalDependencies)
+              </AdditionalDependencies>
+              <AdditionalOptions>$(Conan{name}LinkerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </Link>
+            <Midl>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+            </Midl>
+            <ResourceCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+              <PreprocessorDefinitions>$(Conan{name}PreprocessorDefinitions)
+              %(PreprocessorDefinitions)</PreprocessorDefinitions>
+              <AdditionalOptions>$(Conan{name}CompilerFlags) %(AdditionalOptions)</AdditionalOptions>
+            </ResourceCompile>
+          </ItemDefinitionGroup>
+          <ItemGroup />
+        </Project>
+        """)
+
+    @property
+    def filename(self):
+        return None
+
+    @ staticmethod
+    def _name_condition(settings):
+        toolset = msvs_toolset(settings)
+        if toolset is None:
+            raise ConanException("Undefined Visual Studio version %s" %
+                                 settings.get_safe("compiler.version"))
+
+        props = [("Configuration", settings.build_type),
+                 ("Platform", {'x86': 'Win32',
+                               'x86_64': 'x64'}.get(settings.get_safe("arch"))),
+                 ("PlatformToolset", toolset)]
+
+        name = "".join("_%s" % v for _, v in props if v)
```

Comment:
(This is an edge case, but) If some of them are not required, then the order matters (and several can match). For example:

```
conan install xxxx -g msbuild -s build_type=None
conan install xxxx -g msbuild -s build_type=Debug
```

will get a different order of imports if we do:

```
conan install xxxx -g msbuild -s build_type=Debug
conan install xxxx -g msbuild -s build_type=None
```

and entering with `Configuration=Debug` will **match both `Imports`** (and probably fail to link).

Is it easy to check if `Configuration` is not defined in those `Condition` attributes?